src/lib/Client.ts

```diff
@@ -0,0 +1,57 @@
+import fetch from "node-fetch";
+import ResponseError from "./ResponseError";
+import { APIResponse } from "./types";
+
+class Client {
+  private _apiKey: string;
+
+  constructor(apiKey: string) {
+    if (!apiKey) {
+      throw new Error("No API key provided");
+    }
+
+    this._apiKey = apiKey;
+  }
+
+  public async lookup(ip: string): Promise<APIResponse> {
+    try {
+      const response = await fetch(
+        `https://whois.as207111.net/api/lookup?ip_address=${encodeURI(ip)}`,
+        {
+          method: "GET",
+          headers: {
+            Accept: "application/json",
+            Authorization: `Bearer ${this._apiKey}`,
+          },
+        }
+      );
+
+      if (response.status !== 200) {
+        switch (response.status) {
+          case 503:
+            throw new ResponseError(503, "Database Error");
+
+          case 409:
+            throw new ResponseError(409, "Lookup Error");
```

Comment:
We should use `Conflict`, because it's more generic.

Reply:
the messages were taken from the data you sent on the discord server, I'll change it